Fix criteria mapping type

franmomu opened the pull request:
<!-- Fill in the relevant information below to help triage your pull request. -->

|      Q       |   A
|------------- | -----------
| Type         | bug
| BC Break     | no
| Fixed issues | <!-- use #NUM format to reference an issue -->

#### Summary

<!-- Provide a summary your change. -->

While doing https://github.com/doctrine/mongodb-odm/pull/2596, the SA tool complained about the type of this property. For example in https://github.com/doctrine/mongodb-odm/blob/a51bfc01425d2c30b69f2c5ce13fe0d3c58bbd98/tests/Documents/BlogPost.php#L63
`$criteria` is holding an array that contains a boolean.

It is used in https://github.com/doctrine/mongodb-odm/blob/a51bfc01425d2c30b69f2c5ce13fe0d3c58bbd98/lib/Doctrine/ODM/MongoDB/Persisters/DocumentPersister.php#L762-L767
and 
https://github.com/doctrine/mongodb-odm/blob/a51bfc01425d2c30b69f2c5ce13fe0d3c58bbd98/lib/Doctrine/ODM/MongoDB/Persisters/DocumentPersister.php#L830-L834

so it looks like it could contain any type, not only strings.

Ref: https://github.com/doctrine/mongodb-odm/pull/2596/files#r1441864943

franmomu committed:
Fix criteria mapping type

malarzm reviewed (approved):
Good catch! There's new PHPStan complaint tho :D

franmomu committed:
Ignore psalm issues

These new issues have been reported in https://github.com/vimeo/psalm/issues/10578

franmomu commented:
> Good catch! There's new PHPStan complaint tho :D

I've ignored the complaint since it has been reported in https://github.com/vimeo/psalm/issues/10578

SenseException reviewed (approved):
<![CDATA[Thank you]]>
